lib/ssh_scan/config_loader.rb

```diff
@@ -0,0 +1,31 @@
+require 'logger'
+
+require 'ssh_scan/config_loader/yaml'
+
+module SSHScan
+  module ConfigLoader
+    def self.load(config, class_name = nil, mode = 'file', logger = nil)
```

Comment:
mode seems redundant in this context, this seems like something we can easily detect without requiring the user to specify.

Example:

    case config
    when File
       ...#it's a file
    when String
      if config.include?(".yml") && File.exist?(config)
        ... #it's a path
      else
        ... #it's a string
      end
    else
       raise "unsupported config object type #{config.class}"
    end